tough/src/schema/mod.rs

```diff
@@ -308,7 +311,69 @@ impl Targets {
             expires,
             targets: HashMap::new(),
             _extra: HashMap::new(),
+            delegations: None,
+        }
+    }
+
+    ///Given a target url, returns a reference to the Target struct or error if the target is unreachable
+    pub fn find_target(&self, target_url: &str) -> Result<&Target> {
+        match self.targets.get(target_url) {
+            Some(target) => Ok(target),
+            None => match &self.delegations {
+                None => Err(Error::TargetNotFound {
+                    target_url: target_url.to_string(),
+                }),
+                Some(delegations) => delegations.find_target(target_url),
+            },
+        }
+    }
+
+    ///Given the name of a delegated role, return the delegated role
+    pub fn get_del_role(&self, name: &str) -> Result<&DelegatedRole> {
+        self.delegations.as_ref().unwrap().get_del_role(name)
```

Comment:
Need to do a match case on the Option instead of `unwrap()`. Also something to think about, the return could be an option instead of result, here and in other places. In the standard library, e.g. HashMap `get` returns an `Option` instead of a `Result`. Doesn't seem like a big deal either way though.

```rust
        self.delegations.as_ref().unwrap().get_del_role(name)
```